### PR TITLE
bug: [add more detailed debugging messages to json decoder] (FF-2580)

### PR DESCRIPTION
**Pull request by leoromanovsky**
**motivation**

addressing two bugs:

* http requests going to RAC path instead of UFC
* `sdkVersion` hard-coded to string `sdkVersion` instead of the variable

**additional testing**

* decoding the obfuscated rac

### CODE REVIEW

File: Sources/eppo/UniversalFlagConfig.swift [hunk 1]
```diff
@@ -28,13 +26,13 @@ public struct UniversalFlagConfig : Decodable {
         } catch let error as DecodingError {
             switch error {
             case .typeMismatch(_, let context):
-                throw UniversalFlagConfigError.parsingError("Type mismatch: \(context.debugDescription)")
+                throw UniversalFlagConfigError.parsingError("Type mismatch: \(context.debugDescription) - \(String(describing: context.underlyingError))")
```
Comment (leoromanovsky) [inline Sources/eppo/UniversalFlagConfig.swift:29]: The existing error message does not provide enough information to diagnose the issue. the `underlyingError` should demonstrate the exact object that cannot be decoded.

* https://developer.apple.com/documentation/swift/decodingerror/context/underlyingerror

File: Tests/eppo/UniversalFlagConfigTests.swift [hunk 2]
```diff
@@ -43,6 +40,45 @@ final class UniversalFlagConfigTest: XCTestCase {
         // total shards
         XCTAssertEqual(variationFlag?.totalShards, 10000, "The total shards should be 10000.")
     }
+    
+    func testDecodeObfuscatedUFCConfig() {
```
Comment (leoromanovsky) [inline Tests/eppo/UniversalFlagConfigTests.swift:44]: Added decoding tests for the obfuscated json just to have in-place

File: Tests/eppo/UniversalFlagConfigTests.swift [hunk 3]
```diff
@@ -56,7 +92,7 @@ final class UniversalFlagConfigTest: XCTestCase {
                 return
             }
             XCTAssertEqual(thrownError.errorCode, 101, "Error code should be 101 indicating a JSON parsing issue")
-            XCTAssertEqual(thrownError.localizedDescription, "Data corrupted: The given data was not valid JSON.")
+            XCTAssertTrue(thrownError.localizedDescription.hasPrefix("Data corrupted: The given data was not valid JSON."))
```
Comment (leoromanovsky) [inline Tests/eppo/UniversalFlagConfigTests.swift:95]: this error message got longer now with the `underlyingError` added to it; this is sufficient

File: Sources/eppo/UniversalFlagConfig.swift [hunk 4]
```diff
@@ -17,7 +16,6 @@ public struct UniversalFlagConfig : Decodable {
             let container = try decoder.singleValueContainer()
             let dateStr = try container.decode(String.self)
             guard let date = parseUtcISODateElement(dateStr) else {
-                print(dateStr)
```
Comment (leoromanovsky) [inline Sources/eppo/UniversalFlagConfig.swift:20]: remove stray print message

File: Sources/eppo/EppoClient.swift [hunk 5]
```diff
@@ -58,7 +58,7 @@ public class EppoClient {
         self.assignmentLogger = assignmentLogger
         self.assignmentCache = assignmentCache
         
-        let httpClient = NetworkEppoHttpClient(baseURL: host, sdkKey: sdkKey, sdkName: "sdkName", sdkVersion: "sdkVersion")
+        let httpClient = NetworkEppoHttpClient(baseURL: host, sdkKey: sdkKey, sdkName: "sdkName", sdkVersion: sdkVersion)
```
Comment (yfrancis) [inline Sources/eppo/EppoClient.swift:61]: should we update `sdkName` to also use the constant?

File: Sources/eppo/Obfuscation.swift [hunk 6]
```diff
@@ -21,6 +21,13 @@ func getMD5Hex(_ value: String) -> String {
     return digestData.map { String(format: "%02hhx", $0) }.joined()
 }
 
+func base64Encode(_ value: String) -> String {
```
Comment (yfrancis) [inline Sources/eppo/Obfuscation.swift:24]: is this change necessary to include in this diff?

Comment (leoromanovsky) [reply Sources/eppo/Obfuscation.swift:24]: It was to support additional testing but is not strictly needed.

File: Sources/eppo/EppoClient.swift (hunk 5 above)
Comment (leoromanovsky) [reply Sources/eppo/EppoClient.swift:61]: Ok I will for consistency.

**Review by baronhoot — APPROVED** [review]

**Review by yfrancis — APPROVED** [review]
